litebird_sim/observations.py

```diff
@@ -273,10 +300,86 @@ def n_blocks_time(self):
     def n_blocks_det(self):
         return self._n_blocks_det
 
+    def _make_detector_blocks(self, detectors, comm):
+        """This function distributes the detectors in groups such that each
+        group has same set of attributes specified by the strings in
+        `self._det_block_attributes`. Once the groups are made, number of
+        detector blocks is set to be the total number of detector groups
+        whereas the number of time blocks is computed using the number of
+        detector blocks and the size of `comm` communicator.
+
+        The blocks of detectors are stored in `self.detector_blocks`. It is a
+        dictionary object with the tuple of `self._det_blocks_attributes` values
+        as dictionary keys and the list of detectors corresponding to the key
+        as the dictionary value. This dictionary is sorted so that that the
+        group with largest number of detectors comes first and the one with
+        the least number of detectors, comes last.
```

Comment:
Better:

> This function distributes the detectors in groups such that each group has the same set of attributes specified by the strings in `self._det_block_attributes`. Once the groups are made, the number of detector blocks is set to be the total number of detector groups, whereas the number of time blocks is computed using the number of detector blocks and the size of the `comm` communicator.
The detector blocks are stored in `self.detector_blocks`. This dictionary object has the tuple of `self._det_blocks_attributes` values as dictionary keys and the list of detectors corresponding to the key as dictionary values. This dictionary is sorted so that the group with the largest number of detectors comes first and the one with the fewest detectors comes last.